server/src/main/java/org/elasticsearch/transport/InboundPipeline.java

```diff
@@ -29,7 +30,7 @@ public class InboundPipeline implements Releasable {
     private final InboundDecoder decoder;
     private final InboundAggregator aggregator;
     private final BiConsumer<TcpChannel, InboundMessage> messageHandler;
-    private Exception uncaughtException;
+    protected final AtomicReference<Exception> uncaughtExceptionReference = new AtomicReference<>();
```

Comment:
We had discussion for using AtomicReference in this context. There were concerns about whether this is compatible from a design perspective and we might have to use the eventloop to submit any changes. For pure safety point of view, I think this approach is OK especially since the connection gets closed once this exception is observed. But I'd like to flag it for @Tim-Brooks's opnion. Thanks!

Reply:
We ultimatley went with the [fireExceptionCaught](https://github.com/elastic/elasticsearch/pull/98814/commits/ee41575a4d905a7eece2c92eb14ff5885a64e681) option.